### ISSUE TITLE
`GetPreprocessingSymbolInfo` returns info when inside `#pragma warning` directive

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#72907]
During #66425 we discovered that `CSharpSemanticModel.GetPreprocessingSymbolInfo(IdentifierNameSyntax)` returns valid preprocessing info when invoked on an `IdentifierNameSyntax` inside a `#pragma warning` directive, where the identifier is not a preprocessing symbol.

Upon request of @CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
Can you give an example of what you mean?

**Comment by Rekkonnect** [comment]
Consider the following example
```csharp
#pragma warning disable $$SYMBOL
```

When you invoke the `GetPreprocessingSymbolInfo` on the `IdentifierNameSyntax` containing `SYMBOL`, you get a `PreprocessingSymbolInfo` that has a non-null `PreprocessingSymbolInfo.Symbol`.

**Comment by CyrusNajmabadi** [comment]
Yes.  That seems odd, and likely a bug.  If you want to create a PR on this, i think that would be ok.  We can get compiler to weigh in next week.

**Comment by Rekkonnect** [comment]
Revisiting this after some time, the main issue is the unexpected result from this invocation. We know that preprocessing symbols are names that can be used in preprocessing directives, but such symbols can also be returned in a pragma warning directive. It now seems to me that the hidden intention is that we treat diagnostic code identifiers as preprocessing symbols. The key that is missing here is some way to distinguish that.

The options as far as I can see are:
- introduce a new property in `IPreprocessingSymbol` to denote whether the symbol refers to a diagnostic code or not
- introduce a new symbol kind like `IDiagnosticSymbol`, which I find overkill

Additionally, if that path is taken and diagnostic code identifiers are treated differently, it leaves up room for the numeric literal version of pragma warning directives. For example, in `#pragma warning disable 312`, `312` should be a symbol denoting the diagnostic code CS0312 (I think this is how it works).

**Comment by CyrusNajmabadi** [comment]
> . It now seems to me that the hidden intention is that we treat diagnostic code identifiers as preprocessing symbols.

What is meant by that?
